microservices/first/utils/db_manager.py

```diff
@@ -0,0 +1,127 @@
+    """
+    self.db_file = db_file
+    self.conn = self.create_connection()
+
+  def create_connection(self) -> Optional[sqlite3.Connection]:
+    """Create a database connection to the SQLite database."""
+    conn = None
+    try:
+        conn = sqlite3.connect(self.db_file)
+        return conn
+    except Error as e:
+        print(e)
+    return conn
+
+  def close_connection(self) -> None:
+    """Close the database connection."""
+    if self.conn:
+        self.conn.close()
+
+  def create_deployments_table(self) -> None:
+    """Create the deployments table in the database."""
+    try:
+        c = self.conn.cursor()
+        c.execute(
+                  '''CREATE TABLE IF NOT EXISTS deployments (
+                  id INTEGER PRIMARY KEY, 
+                  repo_name TEXT, 
+                  repo_branch TEXT,
+                  status TEXT DEFAULT 'STARTED')'''
+                )
+    except Error as e:
+        print(e)
+
+  def create_results_table(self) -> None:
+    """Create the results table in the database."""
+    try:
+        c = self.conn.cursor()
+        c.execute(
+            '''CREATE TABLE IF NOT EXISTS results (
+                id INTEGER PRIMARY KEY,
+                deployment_id INTEGER,
+                result TEXT,
+                FOREIGN KEY(deployment_id) REFERENCES deployments(id)
+            )'''
+        )
+    except Error as e:
+        print(e)
+
+  def add_result(self, deployment_id: int, result: str) -> int:
+    """Add a new result to a specific deployment."""
+    sql = '''INSERT INTO results(deployment_id, result) VALUES(?, ?)'''
+    cur = self.conn.cursor()
+    cur.execute(sql, (deployment_id, result))
+    self.conn.commit()
+    return cur.lastrowid
+  
+  def add_results(self, deployment_id: int, results: list) -> None:
+    """Add multiple results to a specific deployment."""
+    sql = '''INSERT INTO results(deployment_id, result) VALUES(?, ?)'''
+    cur = self.conn.cursor()
+    for result in results:
+        cur.execute(sql, (deployment_id, result))
+    self.conn.commit()
+
+  def get_results(self, deployment_id: int) -> list:
+    """Retrieve all results for a specific deployment."""
+    cur = self.conn.cursor()
+    cur.execute("SELECT result FROM results WHERE deployment_id=?", (deployment_id,))
+    results = [row[0] for row in cur.fetchall()]
+    return results
+
+  def add_deployment(self, repo_name: str, repo_branch: str) -> int:
+    """
+    Add a new deployment to the database.
+    
+    param repo_name: The name of the repository for the deployment.
+    param repo_branch: The branch of the repository for the deployment.
+    """
+    sql = '''INSERT INTO deployments(repo_name, repo_branch) VALUES(?,?)'''
+    cur = self.conn.cursor()
+    cur.execute(sql, (repo_name, repo_branch))
+    self.conn.commit()
+    return cur.lastrowid
+
+  def update_deployment_fields(self, deployment_id: int, updates: dict) -> None:
```

Comment:
Can add types to dict

```suggestion
  def update_deployment_fields(self, deployment_id: int, updates: dict[str,str]) -> None:
```